detd/manager.py

```diff
@@ -208,3 +218,32 @@ def update_base_time(self, config):
         safety_margin = multiple * period
 
         config.stream.base_time = (now + ns_until_next_cycle) + safety_margin
+
+    def _get_device_hints(self, config):
+
+        # Get the device suggested default hints
+        hints = self.interface.device.device_hints()
+
+        if config.hints is not None:
```

Comment:
Please move the check to a function implemented in each device plugin. That takes a Hints instance and returns true or false.